src/main/java/net/oauth/jsontoken/AbstractJsonTokenParser.java

```diff
@@ -0,0 +1,229 @@
+/**
+ * Copyright 2010 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package net.oauth.jsontoken;
+
+import com.google.common.base.Preconditions;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+
+import net.oauth.jsontoken.crypto.AsciiStringVerifier;
+import net.oauth.jsontoken.crypto.SignatureAlgorithm;
+import net.oauth.jsontoken.crypto.Verifier;
+
+import org.apache.commons.codec.binary.Base64;
+import org.joda.time.Instant;
+
+import java.security.SignatureException;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * Class that provides common functions
+ * used by {@link JsonTokenParser} and {@link AsyncJsonTokenParser}.
+ */
+public abstract class AbstractJsonTokenParser {
+  protected final Clock clock;
```

Comment:
What code in the implementing classes uses these protected variables? Rather than exposing the fields directly, is it possible to expose protected semantic methods that will encapsulate any logic needed by implementing classes?

Reply:
We can add protected methods if needed, but these variables likely won't be directly used by anything outside of the class. I will make them private.